src/hb-ot-name-table.hh

```diff
@@ -209,7 +220,17 @@ struct name
 
     auto it =
     + nameRecordZ.as_array (count)
-    | hb_filter (c->plan->name_ids, &NameRecord::nameID)
+    | hb_filter ([&] (const NameRecord& namerecord)
```

Comment:
I think you can do this as three separate filters, without needing to define a lambda:
...
| hb_filter (c->plan->name_ids, &NameRecord::nameID)
| hb_filter (c->plan->name_languages, &NameRecord::languageID);

if (!c->plan->name_legacy) {
  it = it | hb_filter(&NameRecord::isUnicode);
}